packages/headless/src/features/search/search-actions.ts

```diff
@@ -160,6 +160,12 @@ export const executeSearch = createAsyncThunk<
       return rejectWithValue(retried.response.error);
     }
 
+    const retriedResponse = retried.response.success;
+    analyticsAction(
+      dispatch,
+      () => getStateAfterResponse(correctedQuery, state, retriedResponse),
+      extra
+    );
```

Comment:
This is where the magic happens.

I'm a redux noob, so it took me way too long to figure out, but I can mock the state that the action expects this way. It's a bit of a dirty fix, but if it needs to be replaced in the future, that should be easy enough.